### ISSUE TITLE
Keep archived object solr index but they don't show up at the default search

### ISSUE THREAD

**Issue by taojing2002** [NCEAS/metacat#1285]
This ticket is related to this ticket:
https://github.com/NCEAS/metacatui/issues/744

Here is the summery from Bryce based on our all-hands meeting:
(Circling back on this). We had a fruitful discussion about this in our CI breakout at the DataONE AHM last week. I'll try to summarize them here but I'd appreciate if others would chime in.

I think we all agreed that @vchendrix's use case is very important:

If a user has a doi and wants to see the old data package, they should be able to access it.

We think "archival" should essentially mean,

From an API perspective: Hide archived records in Solr queries by default, show them when explicitly requested
From a MetacatUI perspective: Only hide archived content in search listings, show elsewhere
The current sense of "archival" is to hide the record from the all queries regardless and this disrupts the notion of DataONE being good for long-term archival (note that overloaded use of the word here).

What does this mean?

The default behavior of the Solr index will be to not show archived content unless explicitly requested by the client. i.e., by passing an archived:true query parameter.
Archived content will no longer be absent from the Solr index but will not be returned by default, which means this is a non-breaking change for clients.
While client tools will not break and will see the same results as they would've prior to this change but they will need updating in order to properly show archived content where appropriate (see below) such as showing dataset landing pages for archived content.
Where will you see archived content in MetacatUI?

Simply: When you know the identifier. This is similar to the common web idiom of "only users with the link can view".

As for MetacatUI, this means:

/catalog: Does not show archived metadata records (current behavior)
/view: Does show archived/non-archived Data Packages (new behavior)
/profile: Does not show archived/non-archived Data Packages (current behavior)
Note: I think we were learning towards /profile showing archived content by default but I think it's more consistent to not show archived content in /profile since it's a public listing just like /catalog. I'm open to more discussion on this.

Why can't we just fall back to the ORE when we run into archived content?

Displaying a human-readable view of a dataset requires the search index. First, to route from the metadata PID the user is requesting a view of to the Data Pakcage(s) it's contained in and, second, to bulk query for the members of the Data Packages and their associated metdata (file names, sizes, etc).

What does ESS-DIVE need to do to archive content?

After these changes, the minimum thing to do to archive a dataset would be to archive just the metadata record. The best thing, as far as the API goes, would be to archive the metadata, the ORE, and all aggregated objects. The above page in your screenshot would show the actual package and not just a broken page.

How do I really hide my content?

After this change, the only option is either:

Remove public read or other permissions from the object(s) in question
Administratively delete objects
I think this is a bit hacky to leave these as the only two options.

Steps forward

Look into how to achieve this with Solr (hide archived content by default)
Update DataONE indexing stack to reflect this change
Update MetacatUI so dataset landing pages like in the screenshot at the top work with the new indexing behavior
Update DataONE API docs

**Comment by taojing2002** [comment]
Dave pointed me a link  which can achieve the goal:  The default behavior of the Solr index will be to not show archived content unless explicitly requested by the client:
https://hpad.dataone.org/dataone_messaging_design
Here is what I did in the solrconfig.xml
```
<str name="archived">archived:false</str>
<lst name="appends">
         <str name="fq">{!v=$archived}</str>
</lst>
```
When we want to show the archived the objects, we need to append this string on our regular query:
```
&archived=archived:true
```
Note: it doesn't work if you just set the `archived:true` since it uses a string replacement mechanism.  You have to append the whole string `&archived=archived:true`.
The `archived` which is on the left side of `=` is the name of a property we defined on the solrconfig.xml file.  It can be anything. I chose this name. If we decided, we should adhere to it otherwise the client's query should be changed. **If you have different idea about the name, please let me know.**
The `archived` which is on the right side of `=` is the solr field name. 

This is the query only shows the objects whose value of `archived` is false:
```
https://valley.duckdns.org/metacat/d1/mn/v2/query/solr/q=id:*&fl=id,archived
```
This is the query only shows the objects whose value of `archived' is true.
```
https://valley.duckdns.org/metacat/d1/mn/v2/query/solr/q=id:*&fl=id,archived&archived=archived:true
```
This the query will shows the objects whose value of `archived' is either true or false.
```
https://valley.duckdns.org/metacat/d1/mn/v2/query/solr/q=id:*&fl=id,archived&archived=archived:true%20OR%20archived:false
```

**Comment by taojing2002** [comment]
The delete method still removes the solr document of the object, right?

**Comment by taojing2002** [comment]
The changed was merged from the feature branch to dev-2.10.
